api/testworkflows/v1/types.go

```diff
@@ -241,6 +241,8 @@ type TestWorkflowPvcConfig struct {
 	Resources *Resources `json:"resources,omitempty" expr:"include"`
 	// Storage class name specifies the name of a StorageClass. More info: https://kubernetes.io/docs/concepts/storage/storage-classes/
 	StorageClassName string `json:"storageClassName,omitempty" expr:"template"`
+	// volume name is used to identify the volume
+	VolumeName string `json:"volumeName,omitempty" expr:"template"`
```

Comment:
Is there a reason to add `volumeName` to it? Often you may specifically not want to attach it as a volume, also attaching it down (to services/parallel steps) would be a bit strange then (as you wouldn't be able to do it manually (access mode), while it's strange to have to create volume there, but not layer below)

Reply:
it's optional, as well as selector, do you prefer to see a pointer?

Reply:
k8s api

	// accessModes contains the desired access modes the volume should have.
	// More info: https://kubernetes.io/docs/concepts/storage/persistent-volumes#access-modes-1
	// +optional
	AccessModes []PersistentVolumeAccessMode `json:"accessModes,omitempty" protobuf:"bytes,1,rep,name=accessModes,casttype=PersistentVolumeAccessMode"`
	// selector is a label query over volumes to consider for binding.
	// +optional
	Selector *metav1.LabelSelector `json:"selector,omitempty" protobuf:"bytes,4,opt,name=selector"`
	// resources represents the minimum resources the volume should have.
	// If RecoverVolumeExpansionFailure feature is enabled users are allowed to specify resource requirements
	// that are lower than previous value but must still be higher than capacity recorded in the
	// status field of the claim.
	// More info: https://kubernetes.io/docs/concepts/storage/persistent-volumes#resources
	// +optional
	Resources VolumeResourceRequirements `json:"resources,omitempty" protobuf:"bytes,2,opt,name=resources"`
	// volumeName is the binding reference to the PersistentVolume backing this claim.
	// +optional
	VolumeName string `json:"volumeName,omitempty" protobuf:"bytes,3,opt,name=volumeName"`
	// storageClassName is the name of the StorageClass required by the claim.
	// More info: https://kubernetes.io/docs/concepts/storage/persistent-volumes#class-1
	// +optional
	StorageClassName *string `json:"storageClassName,omitempty" protobuf:"bytes,5,opt,name=storageClassName"`
	// volumeMode defines what type of volume is required by the claim.
	// Value of Filesystem is implied when not included in claim spec.
	// +optional
	VolumeMode *PersistentVolumeMode `json:"volumeMode,omitempty" protobuf:"bytes,6,opt,name=volumeMode,casttype=PersistentVolumeMode"`

Reply:
Ah, I see, I thought that it's about Pod volume, not Persistent Volume 👍 Looks good then, sorry for interruption! 👍